src/wp-admin/includes/ajax-actions.php

```diff
@@ -4588,6 +4588,39 @@ function wp_ajax_activate_plugin() {
 		wp_send_json_error( $status );
 	}
 
+	/**
+	 * Filters the configuration data for the plugin.
+	 *
+	 * The dynamic portion of the hook name refers to the plugin slug.
+	 *
+	 * How WordPress uses this data may change in the future.
+	 *
+	 * Currently, the user will be redirected to the configuration URL
+	 * after activating the plugin.
```

Comment:
Let's discuss and iterate on improving the filter's description per @costdev [suggestion of focusing it on ""What am I filtering?"](https://core.trac.wordpress.org/ticket/61269#comment:39) rather than how Core will use the configuration data. 

Note: The filter is not driving the behavior, which is controlled by Core. Rather, it's the entry point where a plugin registers their configuration data with the install > activate > onboard workflows.

What's being filtered?

How about this?

```suggestion
	 * Filters the configuration data for the plugin in the install > activate workflow.
	 *
	 * The dynamic portion of the hook name refers to the plugin slug.
```

@costdev what is a better way to describe " install > activate workflow"?

Reply:
Maybe “initial plugin activation workflow”?

Reply:
Adjusted using @afragen suggestion. Is this better and more clear?

Reply:
Thinking more about this .... "activation workflow" might lock into only being for activation. This could be limiting for when the next step is introduced, though one could argue, this is adding the entry point into the next step.

What if it's changed to the "add plugin workflow":

>Filters the configuration data for the add plugin workflow.

This wording aligns well to the Add Plugin UI where the cards and modal are. Then when the 3rd step is introduced in a later release, it could be expanded if needed.

@afragen @costdev and everyone, what do you think?

Reply:
Honestly I think any follow up steps should be handled by the plugin and not core. 

My reasoning is that there can be too many possible next steps. 

Core should handle the most common process and that's the initiation of the onboarding the plugin should be able to handle any subsequent processes.

Reply:
Note that this exact filter is maybe/probably/likely to be used to add a similar link in a plugin's action links on the `Plugins > Installed plugins` screen.

We can always update a filter description later, but something like this may be more fitting for the longer term:

```php
Filters data for a plugin's configuration workflow.
```

That way, it's not about the workflows that lead to this data, but rather the workflows that this data leads to.

Reply:
Aha, yes, that's better and on point @costdev.

Should it use the same language to include onboarding | set up | configuration?

>Filters data for a plugin's onboarding | set up | configuration workflow.

Reply:
Hmm, so technically, "set up" and "configuration" are usually the same, it's just that "set up" is like a "first-time configuration". Even if "configuration" points to more settings and "set up" is just the core configuration, using "configuration" should still work.

I'm mindful that if we add "onboarding" to the main subject of the docblock, then changing it later may be problematic.

We could do something like this maybe?
> Filters the data for the plugin's configuration workflow.
>
> In WordPress 6.5, this can be used to supply data for the plugin's onboarding workflow too.
> See https://core.trac.wordpress.org/ticket/61040 to follow future development on this.

What do you think?

Reply:
Thinking more on this, even if we introduce an onboarding framework later, this filter can be used to supply any URL, so maybe we don't need to detail different use cases here, and just support the provision of a URL to be determined by the plugin author.

Reply:
>Hmm, so technically, "set up" and "configuration" are usually the same, it's just that "set up" is like a "first-time configuration". Even if "configuration" points to more settings and "set up" is just the core configuration, using "configuration" should still work.

The first time I read the onboarding ticket I thought - huh, onboarding is an experience for configuring. I've been using the 3 separate terms for consistency. But to me, they are all forms of configuration.

Thus, I agree with you. It's configuration. Even the "onboarding framework" ticket is about presenting users with the path towards "configuration".

 Now back to this filter's docblock.
 
I'm not sure this part adds enough value to be included:

>In WordPress 6.5, this can be used to supply data for the plugin's onboarding workflow too.
See https://core.trac.wordpress.org/ticket/61040 to follow future development on this.

What if that ticket gets closed such as not enough users want it or a different direction surfaces?

I'm thinking [what you proposed here works well](https://github.com/WordPress/wordpress-develop/pull/6606#discussion_r1617860270).

>Filters the data for the plugin's configuration workflow.

I'll push that now.